src/Diagnostics.AIProjects/SearchAPI/SearchModule/views.py

```diff
@@ -56,33 +60,25 @@ def logger(*args, **kwargs):
         return logger
     return loggingOuter
 
-def getAllProductIds(node):
-    allProductIds = []
-    if type(node).__name__=='dict':
-        for key in node.keys():
-            allProductIds += getAllProductIds(node[key])
-    else:
-        allProductIds.append(node)
-    return allProductIds
-
 # App routes
 cors = CORS(app)
 app.config.from_object("AppConfig.ProductionConfig")
 app.config['CORS_HEADERS'] = 'Content-Type'
 
 @app.before_first_request
 def activate_job():
-    productIds = getAllProductIds(resourceConfig)
-    sah = StorageAccountHelper(loggerInstance)
-    loggerInstance.logInsights("Starting model sync for {0}".format(','.join(productIds)))
-    thread = threading.Thread(target=sah.watchModels, args=(productIds,))
-    thread.start()
-    while True:
-        modelDownloadPending = [sah.firstTime[productId] if productId in sah.firstTime else True for productId in productIds]
-        if any(modelDownloadPending):
-            time.sleep(2)
-        else:
-            break
+    if app.config['MODEL_SYNC_ENABLED']:
+        productIds = getAllProductIds(resourceConfig)
+        sah = StorageAccountHelper(loggerInstance)
+        loggerInstance.logInsights("Starting model sync for {0}".format(','.join(productIds)))
+        thread = threading.Thread(target=sah.watchModels, args=(productIds,))
+        thread.start()
+        while True:
```

Comment:
This should include a fail-safe, say if the loop has gone through 1000 iterations break out with an exception. Just to prevent infinite loops due to logic errors.